source/base/partitioner.cc

```diff
@@ -161,7 +161,13 @@ namespace Utilities
         }
 
       std::vector<types::global_dof_index> first_index (n_procs+1);
-      first_index[0] = 0;
+      // Allow non-zero start index for the vector. send this data to all
+      // processors
+      first_index[0] = local_range_data.first;
+      MPI_Bcast(&first_index[0], sizeof(types::global_dof_index), MPI_BYTE,
+                0, communicator);
```

Comment:
OK, thanks. I have no other comments -- since @Rombur seemed ok with it, feel free to merge whenever you're happy with the current state.